### ISSUE TITLE
Cannot match against compiled rules to improve performance.

### ISSUE THREAD

**Issue by systemxcom** [MITRECND/yaraprocessor#2]
After compiling a large rules set with yarac, getting error in yaraproceesor:
p = Processor(['/etc/yara/sigcomp.yara'])

Error:
yara.SyntaxError: /etc/yara/sigcomp.yara(1): non-ascii character

This compiled ruleset works in yara-python but not in yaraproceesor.

**Comment by wxsBSD** [comment]
Yara processor does not support pre-compiled rules. It shouldn't be that hard to add support for using yara.load() instead of yara.compile(). The reason this isn't implemented is at the time yaraprocessor was written, yara did not support loading compiled rules.

**Comment by dicato** [comment]
@systemxcom thanks for taking the time to create an issue!

@wxsBSD, thanks for responding as well. You are right. This is something I thought of recently since I have not tested yaraprocessor against the latest couple releases of yara.

I'm happy to try and add support for loading from compiled rulesets. I'm hoping the size of the ruleset has no impact on this issue.

**Comment by systemxcom** [comment]
Thank you.  That would be great.  I am attempting to run in near realtime and that would be very helpful.  Could this be done by implementing yara.load() in the yaraprocessor.py?

**Comment by wxsBSD** [comment]
Yes, that is exactly what it needs to do. Though we should probably make it clear that you want to load rules instead of compile them when passing the path in.

**Comment by systemxcom** [comment]
Roger that. I will look at that and see if it will work. But will await for update for production.  Thank you so much.

**Comment by systemxcom** [comment]
When: self._rules = yara.load(filepaths=self._rule_files)
Error: TypeError: load() takes no keyword arguments

**Comment by dicato** [comment]
Try using:

```
self._rules = yara.load(self._rule_files)
```

I'll likely spend some time with this over the weekend.

**Comment by systemxcom** [comment]
File "/etc/yara/yaraprocessor/yaraprocessor.py", line 105, in **init**
    self._rules = yara.load(self._rule_files)
TypeError: load() takes 1 argument

Looks like there needs to be a re-config of the rules parser @: def _prepare_rules(self, rules):

**Comment by dicato** [comment]
In case you missed the notifications, I pushed a PR to address this. See #3

**Comment by systemxcom** [comment]
Speed has increased dramatically.  Great work! However, not sure if it is how I am using it, but when using yara signatures, compiled or not compiled, it identifies/matches strings inside of quotes but will not match on bracketed signatures. e.g. "$m4 = "CDllUninstall" ascii wide fullword" will match.  $dec_2 = {8b 4d f8 8b 16 88 04 11 8b 06 41 89 4d f8 c6 04 01 00 43 3b 5d fc} will not match.

I am running it in raw mode using p.analyze().  Also for my clarification, what is the arg structure that you use to access the packet buffering features with this new compile feature added?  Thank you!

**Comment by dicato** [comment]
@systemxcom are you using yaraprocessor directly, or through chopshop?

**Comment by systemxcom** [comment]
Directly.  Here is how I am using it:
p = Processor(['/etc/yara/sigcomp.yara'], compiled=True, processing_mode='sliding_window', buffer_size=1024, window_step=1)

**Comment by dicato** [comment]
That looks correct to me, and matches [how I instantiate a processor](https://github.com/MITRECND/yaraprocessor/blob/20e4122c05bc8d9bbb8d93b17fbdd2f43755d1c4/yaraprocessor.py#L378) when calling `yaraprocessor.py` as a script during testing.

Are you seeing some unexpected behavior?

In regards to quoted signatures versus bracketed, I'll have to dig into it a bit. Yaraprocessor passes loading rules directly into yara, so I suspect it is not directly related to yararprocessor. I'd like to confirm this though.

**Comment by systemxcom** [comment]
Yes I just confirmed that.  Sorry to cloud the issue here.  So I went back to yara and the condition exists there too so let me run that down.  Not a yaraprocessor issue. No unexpected behavior.  It is just unexpectedly fast!  So I will chase down the signatures in {} on the yara side and get a full performance test underway. And thanks for the example and for the great work to get it sorted on the compiled sigs.
